internal/clients/ec2/ec2_client.go

```diff
@@ -34,10 +34,17 @@ func NewEC2Client(ctx context.Context) *Client {
 	cache := aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(
 		config.AWS.Key, config.AWS.Secret, config.AWS.Session))
 
-	c.ec2 = ec2.New(ec2.Options{
-		Region:      config.AWS.Region,
-		Credentials: cache,
-	})
+	if region[0] == "" {
+		c.ec2 = ec2.New(ec2.Options{
+			Region:      config.AWS.Region,
+			Credentials: cache,
+		})
```

Comment:
having this on top of the service makes it bit clearer what are the default values (especially when we want to get rid of it in the future IMHO.
```suggestion
	if region == "" {
	  region = config.AWS.Region
    }
```